DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs

```diff
@@ -1011,6 +1072,70 @@ private void ConnectionManager_WelcomeMessageReceived(object sender, EventArgs e
             gameCheck.InitializeService(gameCheckCancellation);
         }
 
+        private void ConnectionManager_PrivateCTCPReceived(object sender, PrivateCTCPEventArgs e)
+        {
+            foreach (CommandHandlerBase cmdHandler in ctcpCommandHandlers)
+            {
+                if (cmdHandler.Handle(e.Sender, e.Message))
+                    return;
+            }
+
+            Logger.Log("Unhandled private CTCP command: " + e.Message + " from " + e.Sender);
+        }
+
+        private void HandleGameInviteCommand(string sender, string argumentsString)
+        {
+            // arguments are semicolon-delimited
+            var arguments = argumentsString.Split(';');
+
+            // we expect to be given a game name and optionally a password
+            if (arguments.Length < 1 || arguments.Length > 2)
+                return;
+
+            string roomName = arguments[0];
+            string password = (arguments.Length == 2) ? arguments[1] : string.Empty;
+
+            if (!CanReceiveInvitationMessagesFrom(sender))
+                return;
+
+            var gameIndex = lbGameList.HostedGames.FindIndex(hg => hg.RoomName == roomName);
+
+            if (!CanJoinGameByIndex(gameIndex))
+            {
+                // let the host know that we can't accept
+                // note this is not reached for the rejection case
+                connectionManager.SendCustomMessage(new QueuedMessage("PRIVMSG " + sender + " :\u0001" +
+                    ProgramConstants.GAME_INVITATION_FAILED_CTCP_COMMAND + "\u0001",
+                    QueuedMessageType.CHAT_MESSAGE, 0));
+
+                return;
+            }
+
+            var inviteMessageBox = XNAMessageBox.ShowYesNoDialog(WindowManager, "Invitation received",
+                sender + " wants you to join their game (" + roomName + ").  Accept?");
```

Comment:
This will get annoying quickly but I've been informed that you're already working on a rework for that based on private messages notifications.

![Screenshot I received from @dkeetonx ](https://media.discordapp.net/attachments/680366050642100238/716270296226857021/unknown.png)

This one is pretty nice, my suggestion is to tweak margins so that it stays uniform for all elements and implement in that way that it would be reusable for more notifications of different types that could be implemented later (f.ex. "Ask to Join" notification), kind of like `XNAMessageBox.ShowYesNoDialog` is made.

Reply:
I've made the new ChoiceNotificationBox class (shows at top-left of the screen), mostly by copying code from PrivateMessageNotificationBox.  Considered adding a common NotificationBox base class for both but decided against as no controls are shared and pretty much all the functions are slightly different (PrivateMessageNotificationBox is used as single-instance, where ChoiceNotificationBox is stacked infinitely).

![Invitation](https://user-images.githubusercontent.com/30480208/83349294-a8d9f900-a32b-11ea-9b83-c2a69d3fead4.png)

Reply:
The width of the thing is a bit obnoxious but it looks like we don't have word-wrapping so I want to support both long player names and - given it's generic - long other "choice texts".

Reply:
The popup dialog that advises the host of invitation failure has been retained.  Reasoning is the host will want to know ASAP and obviously when an invitation has failed so they can do something about it.

![Invitation failed](https://user-images.githubusercontent.com/30480208/83349519-631e3000-a32d-11ea-9316-eded2b609bb7.png)

Reply:
Good job, nice one with having them on the left, I thought about that but didn't write that idea. I would streamline the code for both anyways so that it would be kind of a generic notification framework for anything needed, I think it would be more logical for PM notifications to also stack, but you do you. Might be the feature for a separate pull request.

You could save space by doing something like this:

```
            GAME INVITE            
    From: username here        
    [Accept]            [Reject]    
```

Reply:
Restyled as specified, except kept the Ignore button instead of Reject as Reject may imply that the host will be notified of the rejection (which they won't).

Reply:
One more alternative if preferred for clarity:

`            GAME INVITE            
    Join fonger's game?       
    [Yes]                   [No]   `

Reply:
After discussion in discord, latest commit implements this notification: 
![image](https://user-images.githubusercontent.com/30480208/83576379-8f060500-a529-11ea-95d1-c63ca2f680ea.png)